### PR TITLE
Add retry support for blocking LoadBalancer

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**
Fixes gh-829

### CODE REVIEW

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 1]
```diff
@@ -453,12 +453,27 @@ set the `spring.cloud.loadbalancer.ribbon.enabled` property to `false`.
...
-You can use `client.ribbon.MaxAutoRetries`, `client.ribbon.MaxAutoRetriesNextServer`, and `client.ribbon.OkToRetryOnAllOperations` properties.
+
 If you would like to disable the retry logic with Spring Retry on the classpath, you can set `spring.cloud.loadbalancer.retry.enabled=false`.
+
+If you would like to implement a `BackOffPolicy` in your retries, you need to create a bean of type `LoadBalancedRetryFactory` and override the `createBackOffPolicy()` method.
+
+===== Ribbon-based retries
+
+For the Ribbon-backed implementation, the load-balanced `RestTemplate` honors some of the Ribbon configuration values related to retrying failed requests.
+You can use `client.ribbon.MaxAutoRetries`, `client.ribbon.MaxAutoRetriesNextServer`, and `client.ribbon.OkToRetryOnAllOperations` properties.
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:464]: Add "the" after "use"

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 2]
```diff
@@ -453,12 +453,27 @@ set the `spring.cloud.loadbalancer.ribbon.enabled` property to `false`.
...
+
 See the https://github.com/Netflix/ribbon/wiki/Getting-Started#the-properties-file-sample-clientproperties[Ribbon documentation] for a description of what these properties do.
 
-If you would like to implement a `BackOffPolicy` in your retries, you need to create a bean of type `LoadBalancedRetryFactory` and override the `createBackOffPolicy` method:
+===== Spring Cloud LoadBalancer-based retries
+
+For the Spring Cloud LoadBalancer-backed implementation, you can set:
+
+- `spring.cloud.loadbalancer.retry.maxRetriesOnSameServiceInstance` - indicates how many times a request should be retried on the same ServiceInstance (counted separately for every selected instance)
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:472]: Make "ServiceInstance" be code (add backticks around it.

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 3]
```diff
@@ -453,12 +453,27 @@ set the `spring.cloud.loadbalancer.ribbon.enabled` property to `false`.
...
+- `spring.cloud.loadbalancer.retry.maxRetriesOnNextServiceInstance` - indicates how many times a request should be retried a newly selected ServiceInstance
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:473]: Make "ServiceInstance" be code (add backticks around it.

**Review by Buzzardo — CHANGES_REQUESTED** [review]

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/blocking/retry/BlockingLoadBalancedRetryPolicy.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+ * to achieve feature-parity.
+ *
+ * @author Olga Maciaszek-Sharma
+ * @since 2.2.6
+ */
+public class BlockingLoadBalancedRetryPolicy implements LoadBalancedRetryPolicy {
+
+	private final LoadBalancerRetryProperties retryProperties;
+
+	private final ServiceInstanceChooser loadBalancerClient;
```
Comment (spencergibb) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/blocking/retry/BlockingLoadBalancedRetryPolicy.java:38]: field name? `serviceInstanceChooser`

**Review by spencergibb — APPROVED** [review]

**Pull request merged**